src/storage/src/source/reclock.rs

```diff
@@ -185,29 +186,64 @@ where
         // We need to compute the answer by iterating over the consolidated remap trace which will
```

Comment:
I think it would be prudent to mention here a poignant detail, that is depended on by all the below logic:

"Note that D is reclockable _only if it is not beyond the source_upper`, which we check above. Otherwise, we cannot make make conclusions about what time it maps to that can't be later violated by the presence of more data for FromTimes that are known to be greater than D."

(please check my logic; its possible that i am being too detailed here; in some sense, the given remap collections can't really exists without the uppers being pushed ahead as well, but i want to drive home the point that `D` has to a time known to be closed, otherwise we can't reclock it. perhaps we can come up with an example.

Reply:
That is correct and I will add this comment, being more explicit doesn't hurt. The tests already test for this case because it's when we return `ReclockError::BeyondUpper(..)`